library/src/interactor_impl.cxx

```diff
@@ -764,4 +766,87 @@ void interactor_impl::UpdateRendererAfterInteraction()
 {
   this->Internals->Style->UpdateRendererAfterInteraction();
 }
+
+//----------------------------------------------------------------------------
+interactor& interactor_impl::command(const std::string& commandString)
+{
+  // Split command with space
+  std::string temp;
+  std::stringstream stringstream{ commandString };
+  std::vector<std::string> results;
+  while (std::getline(stringstream, temp, ' '))
+  {
+    results.push_back(temp);
+  }
+
+  size_t nTokens = results.size();
+  if (nTokens > 3 || nTokens == 0)
+  {
+    log::error("Command: \"", commandString, "\" contains incorrect number of tokens, ignoring");
+    return *this;
+  }
+
+  std::string command = results[0];
+  std::string option = nTokens >= 1 ? results[1] : "";
+  std::string value = nTokens >= 2 ? results[2] : "";
```

Comment:
```suggestion
  const std::regex commandRE("(\\S+)"          // command: non-spaces
                             "(?:\\s+(\\S+))?" // maybe option: non-spaces after some space(s)
                             "(?:\\s+(.+))?"   // maybe value: anything after some space(s)
  );
  std::smatch match;
  if (std::regex_match(commandString, match, commandRE))
  {
    const std::string command = match[1];
    const std::string option = match[2];
    const std::string value = match[3];
```
maybe regex is overkill but relying on raw `getline` feels a bit optimistic

Reply:
reimplemented (without regex)